packages/workspace/src/providers/index.ts

```diff
@@ -32,30 +33,11 @@ const createAffineSocketIOProvider: DocProviderCreator = (
   doc,
   { awareness }
 ): AffineSocketIOProvider => {
-  const provider = new SocketIOProvider(
-    runtimeConfig.serverUrlPrefix + '/',
-    id,
-    doc,
-    {
-      awareness,
-    }
-  );
+  const dataSource = createAffineDataSource(id, doc, awareness);
   return {
     flavour: 'affine-socket-io',
-    passive: true,
-    get connected() {
-      return provider.connected;
-    },
-    connect: () => {
-      provider.connect();
-    },
-    disconnect: () => {
-      provider.disconnect();
-    },
-    cleanup: () => {
-      provider.destroy();
-    },
-  } satisfies AffineSocketIOProvider;
+    ...createLazyProvider(doc, dataSource),
```

Comment:
For IDB, I am a little worried that the provider may has issues on frequent connect/disconnect for the same provider  instance and the cleanup logic may conflicts with the newly connection logic, thus what I did is to create a new instance on re-connection.

Reply:
let's see what will happen then